alf/algorithms/agent_helpers.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2020 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Some helper functions for constructing an Agent instance."""
+
+import os
+
+import torch
+
+import alf
+from alf.algorithms.rl_algorithm import RLAlgorithm
+from alf.data_structures import LossInfo
+from alf.utils.math_ops import add_ignore_empty
+
+
+class AgentStateSpecs(object):
+    def __init__(self):
```

Comment:
Is it possible to make it a more general helper class for future agents beyond the current one? Say users might want to create their own agent with different AgentState.

Reply:
I guess you mean with different algorithms. You need to specify how they interact with each other anyway, for example, one upstream algorithm will change the inputs of a downstream algorithm. So __init__, rollout_step, and train_step have to be reimplemented for a new agent.